From 05dc8e02af317900a568401b6e8568a76c126575 Mon Sep 17 00:00:00 2001
From: Vladimir Ermakov <vooon341@gmail.com>
Date: Wed, 18 Jun 2014 16:04:58 +0400
Subject: [PATCH 7/7] Use external script for pymavlink installation (as
 catkin).

Some code from catkin build system used to fix installation process.
---
 pymavlink/CMakeLists.txt                 | 28 +++++++++++++++++++++++-----
 pymavlink/python_distutils_install.sh.in | 22 ++++++++++++++++++++++
 2 files changed, 45 insertions(+), 5 deletions(-)
 rewrite pymavlink/CMakeLists.txt (99%)
 create mode 100755 pymavlink/python_distutils_install.sh.in

diff --git a/pymavlink/CMakeLists.txt b/pymavlink/CMakeLists.txt
dissimilarity index 99%
index 5ba1175..7fb27ed 100644
--- a/pymavlink/CMakeLists.txt
+++ b/pymavlink/CMakeLists.txt
@@ -1,5 +1,23 @@
-find_package(PythonInterp)
-
-set(SETUP_PY "${CMAKE_CURRENT_SOURCE_DIR}/setup.py")
-install(CODE "execute_process(COMMAND \"${PYTHON_EXECUTABLE}\" \"${SETUP_PY}\" build --build-base \"${CMAKE_CURRENT_BINARY_DIR}/pybuild\" install --install-layout deb --prefix \"${CMAKE_INSTALL_PREFIX}\"
-                WORKING_DIRECTORY \"${CMAKE_CURRENT_SOURCE_DIR}\")")
+# pymavlink support's python2 only
+set(PythonInterp_FIND_VERSION "2")
+find_package(PythonInterp REQUIRED)
+
+# part of catkin/python.cmake
+set(enable_setuptools_deb_layout OFF)
+if(EXISTS "/etc/debian_version")
+  set(enable_setuptools_deb_layout ON)
+endif()
+option(SETUPTOOLS_DEB_LAYOUT "Enable debian style python package layout" ${enable_setuptools_deb_layout})
+
+if(SETUPTOOLS_DEB_LAYOUT)
+  message(STATUS "Using Debian Python package layout")
+  set(SETUPTOOLS_ARG_EXTRA "--install-layout=deb")
+else()
+  message(STATUS "Using default Python package layout")
+endif()
+
+# prepare installation script
+set(INSTALL_SCRIPT ${CMAKE_CURRENT_BINARY_DIR}/python_distutils_install.sh)
+configure_file(python_distutils_install.sh.in ${INSTALL_SCRIPT} @ONLY)
+
+install(CODE "execute_process(COMMAND \"${INSTALL_SCRIPT}\" WORKING_DIRECTORY \"${CMAKE_CURRENT_SOURCE_DIR}\")")
diff --git a/pymavlink/python_distutils_install.sh.in b/pymavlink/python_distutils_install.sh.in
new file mode 100755
index 0000000..b39c2ed
--- /dev/null
+++ b/pymavlink/python_distutils_install.sh.in
@@ -0,0 +1,22 @@
+#!/bin/sh -x
+# Modified version of catkin template/python_distutils_install.sh.in
+
+if [ -n "$DESTDIR" ] ; then
+    case $DESTDIR in
+        /*) # ok
+            ;;
+        *)
+            /bin/echo "DESTDIR argument must be absolute... "
+            /bin/echo "otherwise python's distutils will bork things."
+            exit 1
+    esac
+    DESTDIR_ARG="--root=$DESTDIR"
+fi
+
+/usr/bin/env \
+    "@PYTHON_EXECUTABLE@" \
+    "@CMAKE_CURRENT_SOURCE_DIR@/setup.py" \
+    build --build-base "@CMAKE_CURRENT_BINARY_DIR@/pybuild" \
+    install \
+    $DESTDIR_ARG \
+    @SETUPTOOLS_ARG_EXTRA@ --prefix="@CMAKE_INSTALL_PREFIX@"
-- 
1.9.1

